crates/topos-uci/src/certificate_id.rs

```diff
@@ -60,6 +60,23 @@ impl TryFrom<Vec<u8>> for CertificateId {
     }
 }
 
+impl From<String> for CertificateId {
+    fn from(input: String) -> Self {
+        let id = if let Some(stripped) = input.strip_prefix("0x") {
+            hex::decode(stripped).unwrap_or_else(|_| vec![0u8; CERTIFICATE_ID_LENGTH])
+        } else {
+            hex::decode(&input).unwrap_or_else(|_| vec![0u8; CERTIFICATE_ID_LENGTH])
+        };
+
+        let id_array: [u8; CERTIFICATE_ID_LENGTH] = id
+            .as_slice()
+            .try_into()
+            .unwrap_or([0u8; CERTIFICATE_ID_LENGTH]);
+
+        CertificateId { id: id_array }
```

Comment:
Instead of creating a new `from` implementation, you can refactor the `TryFrom<Vec<u8>>` like that:

```rust
const HEX_CERTIFICATE_ID_LENGTH: usize = 64;

impl TryFrom<&[u8]> for CertificateId {
    type Error = Error;

    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
        let value = if value.starts_with(b"0x") {
            &value[2..]
        } else {
            value
        };

        let length = value.len();

        if length != CERTIFICATE_ID_LENGTH && length != HEX_CERTIFICATE_ID_LENGTH {
            return Err(Error::ValidationError);
        }

        let mut id = [0; CERTIFICATE_ID_LENGTH];

        if length == HEX_CERTIFICATE_LENGTH {
            let value = hex::decode(value).map_err(|_| Error::ValidationError)?;

            id.copy_from_slice(&value[..])
        } else {
            id.copy_from_slice(value);
        }

        Ok(Self { id })
    }
}
      
```